impl/src/main/java/org/jboss/weld/bean/proxy/ProxyInstantiator.java

```diff
@@ -81,24 +81,14 @@ private Factory() {
         }
 
         /**
-         * Obtains new ProxyInstantiator based on given {@link WeldConfiguration}.
+         * Obtains a ProxyInstantiator instance, {@link ConfigurationKey#PROXY_INSTANTIATOR} is not taken into account.
```

Comment:
> `ConfigurationKey#PROXY_INSTANTIATOR` is not taken into account

I'm not sure it's correct to completely disregard this option. I know it's for testing and such only but maybe that's why it should work as an "override all" option?

Reply:
The problem with custom `ProxyInstantiator` is that it might not be available in all deployments. 

For sure this fix does not cover 100% of use cases but should work in most real scenarios.

Reply:
Actually, disregard this comment about disregarding the config option. If I think about it, there is no point in enabling the option in case of these "combined" deployments.